scarlet2/nn.py

```diff
@@ -0,0 +1,156 @@
+# -------------------------------------------------- #
+# This class is used to calculate the gradient       #
+# of the log-probability via calling the nn prior    #
+# ScoreNet model. A custom vjp is created to return  #
+# the nn prior when calling jax.grad()               #
+# -------------------------------------------------- #
+
+try:
+    import numpyro.distributions as dist
+    import numpyro.distributions.constraints as constraints
+    from galaxygrad import HSC_ScoreNet32, HSC_ScoreNet64, ZTF_ScoreNet32, ZTF_ScoreNet64,HSC_LogNet32, HSC_LogNet64# (https://pypi.org/project/galaxygrad/0.0.19)
+
+except ImportError:
+    raise ImportError("scarlet2.nn requires numpyro and galaxygrad=0.0.19")
+
+import jax.numpy as jnp
+import jax.scipy as jsp
+from jax import custom_vjp
+from jax import vjp
+
+
+def pad_fwd(x, trained_model):
```

Comment:
`trained_model` should not omitted here. It's used to get the model from its name given the data shape, but when the user sets the specific galaxygrad model for a specific source size already, this isn't necessary.

Reply:
Instead, pass a `max_size` argument, which determines the full image size that was used for the score model.

Reply:
this has been altered as suggested in nn.py